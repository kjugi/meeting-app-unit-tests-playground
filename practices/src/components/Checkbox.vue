<template>
  <div class="checkbox">
    <label :for="id">
      {{ label }}
    </label>

    <input
      :id="id"
      v-model="localValue"
      type="checkbox"
    />
  </div>
</template>

<script>
export default {
  props: {
    id: {
      type: String,
      required: true
    },
    label: {
      type: String,
      required: true
    },
    value: {
      type: Boolean,
      required: true
    }
  },
  computed: {
    localValue: {
      get () {
        return this.value
      },
      set (val) {
        this.$emit('input', val)
        this.$emit('change')
      }
    }
  }
}
</script>

<style>
.checkbox {
  margin-bottom: 10px;
}
</style>
