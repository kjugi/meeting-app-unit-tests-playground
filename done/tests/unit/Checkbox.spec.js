import { mount } from '@vue/test-utils'
import Checkbox from '@/components/Checkbox.vue'

describe('Checkbox.vue', () => {
  it('emit events when change checked value', () => {
    const wrapper = mount(Checkbox, {
      propsData: {
        id: 'checkbox1',
        label: 'label prop',
        value: false
      }
    })

    wrapper.find('#checkbox1').setChecked()

    expect(wrapper.emitted('change')).toBeDefined()
    expect(wrapper.emitted('change').length).toBe(1)
    expect(wrapper.emitted('change')[0]).toEqual([])
    expect(wrapper.emitted('input')).toBeDefined()
    expect(wrapper.emitted('input').length).toBe(1)
    expect(wrapper.emitted('input')[0][0]).toEqual(true)
  })

  it('default component matches snapshot', () => {
    const wrapper = mount(Checkbox, {
      propsData: {
        id: 'checkbox1',
        label: 'label prop',
        value: false
      }
    })

    expect(wrapper).toMatchSnapshot()
  })
})
