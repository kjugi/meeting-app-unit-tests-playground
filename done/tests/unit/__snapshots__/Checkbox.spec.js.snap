// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Checkbox.vue default component matches snapshot 1`] = `
<div class="checkbox"><label for="checkbox1">
    label prop
  </label> <input id="checkbox1" type="checkbox"></div>
`;
